File: Swift-C++-Testsuite/test/basicOperators/C++/ComparisonTuple.cpp
#include <iostream>
#include <string>
#include <tuple>

int main() {
    bool a = std::make_tuple(41, "zebra") < std::make_tuple(42, "apple");
    bool b = std::make_tuple(3, "apple") < std::make_tuple(3, "bird");
    bool c = std::make_tuple(4, "dog") == std::make_tuple(4, "dog");

    return 0;
}

File: Swift-C++-Testsuite/test/basicOperators/swift/ComparisonTuple.swift
@main
struct MyMain {

    static func main() {
        var a = (41, "zebra") < (42, "apple")  // true because 1 is less than 2; "zebra" and "apple" aren't compared
        var b = (3, "apple") < (3, "bird")  // true because 3 is equal to 3, and "apple" is less than "bird"
        var c = (4, "dog") == (4, "dog")  // true because 4 is equal to 4, and "dog" is equal to "dog"
    }

}
